xmodule/modulestore/split_mongo/caching_descriptor_system.py

```diff
@@ -167,12 +170,83 @@ def xblock_from_json(self, class_, course_key, block_key, block_data, course_ent
             # most recent retrieval is most likely the right one for next caller (see comment above fn)
             self.course_entry = CourseEnvelope(course_entry_override.course_key, self.course_entry.structure)
 
-        definition_id = block_data.definition
+        # Construct the Block Usage Locator:
+        block_locator = course_key.make_usage_key(
+            block_type=block_key.type,
+            block_id=block_key.id,
+        )
 
-        # If no usage id is provided, generate an in-memory id
-        if block_key is None:
-            block_key = BlockKey(block_data.block_type, LocalId())
+        definition_id = block_data.definition or LocalId()
```

Comment:
```suggestion
        # If no definition id is provided, generate an in-memory id
        definition_id = block_data.definition or LocalId()
```

We can move this comment here from `_init_field_data_for_block`.

Reply:
Thanks @Agrendalath! I am going to continue work on this today and hope to have all the tests passing soon. I appreciate the help and the feedback!